src/promptflow/promptflow/_sdk/_pf_client.py

```diff
@@ -243,9 +250,47 @@ def connections(self) -> ConnectionOperations:
         """Connection operations that can manage connections."""
         if not self._connections:
             self._ensure_connection_provider()
-            self._connections = get_connection_operation(self._connection_provider, self._credential)
+            self._connections = PFClient._build_connection_operation(
+                self._connection_provider,
+                self._credential,
+                bonded_user_agent=self._bonded_user_agent,
+            )
         return self._connections
 
+    @staticmethod
+    def _build_connection_operation(connection_provider: str, credential=None, user_agent: str = None, **kwargs):
+        """
+        Build a ConnectionOperation object based on connection provider.
+
+        :param connection_provider: Connection provider, e.g. local, azureml, azureml://subscriptions..., etc.
+        :type connection_provider: str
+        :param credential: Credential when remote provider, default to chained credential DefaultAzureCredential.
+        :type credential: object
+        :param user_agent: User Agent
+        :type user_agent: str
+        """
+        if connection_provider == ConnectionProvider.LOCAL.value:
+            from promptflow._sdk.operations._connection_operations import ConnectionOperations
+
+            logger.debug("PFClient using local connection operations.")
+            connection_operation = ConnectionOperations(**kwargs)
+        elif connection_provider.startswith(ConnectionProvider.AZUREML.value):
+            from promptflow._sdk.operations._local_azure_connection_operations import LocalAzureConnectionOperations
+
+            logger.debug(f"PFClient using local azure connection operations with credential {credential}.")
+            if user_agent is None:
+                connection_operation = LocalAzureConnectionOperations(connection_provider, credential=credential)
+            else:
+                connection_operation = LocalAzureConnectionOperations(connection_provider, user_agent=user_agent)
+        else:
+            error = ValueError(f"Unsupported connection provider: {connection_provider}")
+            raise UserErrorException(
+                target=ErrorTarget.CONTROL_PLANE_SDK,
+                message=str(error),
```

Comment:
Do we need to use the **message format**? That way, we can see the message information in Telemetry.
@wangchao1230